src/main/java/hudson/plugins/analysis/core/HealthAwareRecorder.java

```diff
@@ -342,15 +372,14 @@ protected Object readResolve() {
     }
 
     @Override
-    public final boolean perform(final AbstractBuild<?, ?> build, final Launcher launcher,
-            final BuildListener listener) throws InterruptedException, IOException {
+    public void perform(Run<?, ?> run, FilePath workspace, Launcher launcher, TaskListener listener)
```

Comment:
I do not think this breaks binary compatibility, since both overloads are overrides.